### PR TITLE
cert-manager: upgrade to 1.3.1 on build01

### PR DESCRIPTION
**Pull request by hongkailiu**
https://issues.redhat.com/browse/DPTP-2046

- have to do 0.15.1 to 0.16 to 1.3.1: advised by the cert-manager's dev. Cannot to 1.3.1 directly.
- i tested with a new domain `build01.cert-manager.ci.openshift.org` and the certs are generated correctly.
- pods are running fine with 1.3.1

I will stop myself here and wait for a couple weeks to see if the renewals of other domains in production go well.
After that, build02, and app.ci.
I will remove the underscore in `clusters/build-clusters/common/_cert-manager.yaml` after those steps.

/cc @openshift/openshift-team-developer-productivity-test-platform

### CODE REVIEW

File: clusters/build-clusters/01_cluster/cert-manager/cert-manager.yaml [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+      app.kubernetes.io/instance: cert-manager
+      app.kubernetes.io/name: webhook
+  template:
+    metadata:
+      labels:
+        app: webhook
+        app.kubernetes.io/component: webhook
+        app.kubernetes.io/instance: cert-manager
+        app.kubernetes.io/name: webhook
+    spec:
```
Comment (alvaroaleman) [inline clusters/build-clusters/01_cluster/cert-manager/cert-manager.yaml:124]: This needs anti affinity

File: clusters/build-clusters/01_cluster/cert-manager/cert-manager.yaml [hunk 2]
```diff
@@ -0,0 +1,175 @@
+apiVersion: apps/v1
```
Comment (alvaroaleman) [inline clusters/build-clusters/01_cluster/cert-manager/cert-manager.yaml:1]: Why do we have a cert-manager.yaml and a _cert-manager.yaml?

Comment (hongkailiu) [reply clusters/build-clusters/01_cluster/cert-manager/cert-manager.yaml:1]: Those are taken from upstream.
We split into 2 files before the shared part is in common folder (CDs and RBACs etc) and the customization for each cluster (mainly on deployment, like anti-affinity you suggested above) is in its own folder.

File: clusters/build-clusters/01_cluster/cert-manager/cert-manager.yaml (hunk 1 above)
Comment (hongkailiu) [reply clusters/build-clusters/01_cluster/cert-manager/cert-manager.yaml:124]: Added.

File: clusters/build-clusters/01_cluster/cert-manager/cert-manager.yaml (hunk 2 above)
Comment (hongkailiu) [reply clusters/build-clusters/01_cluster/cert-manager/cert-manager.yaml]: Renamed the one in cluster folder.

**Comment by alvaroaleman** [comment]
> clusters/build-clusters/common/_cert-manager.yaml

Why does this file exist? Its not going to be applied, is it?

**Comment by hongkailiu** [comment]
> > clusters/build-clusters/common/_cert-manager.yaml
> 
> Why does this file exist? Its not going to be applied, is it?

It is not.
See the PR description: I will remove the underscore when upgrade is done.

**Comment by alvaroaleman** [comment]
> See the PR description: I will remove the underscore when upgrade is done.

Well, why not simply create a second PR that replaces the current file with the file that now has an underscrore instead of adding an underscore file that does nothing?

**Review by alvaroaleman — APPROVED** [review]

**Pull request merged**